src/duckling/migration/migration.service.ts

```diff
@@ -72,16 +73,49 @@ export class MigrationService {
             throw new Error(`You should update duckling. The project expects editor version ${versionFile.editorVersion}. The current version is  ${EDITOR_VERSION}`);
         }
 
+        await this._addMissingEditorMigrations(versionFile, this._findMissingEditorMigrations(versionFile.mapMigrations), versionFileName);
+
         return {
             mapMigrations : versionFile.mapMigrations,
             mapVersion : versionFile.projectVersion
         }
     }
 
+    private async _addMissingEditorMigrations(versionFile : VersionFile, missingMigrations : MapMigration[], versionFileName : string) {
+        versionFile.mapMigrations = versionFile.mapMigrations.concat(missingMigrations);
+        for (let migration of missingMigrations) {
+            versionFile.projectVersion = incrementMajorVersion(versionFile.projectVersion);
+            migration.updateTo = versionFile.projectVersion;
```

Comment:
You should avoid mutating the migrations. This is likely to cause a problem in the future.